src/extension.ts

```diff
@@ -28,86 +21,91 @@ interface LanguagesData {
+  return vscode.commands.registerCommand("lama2.Lama2Examples", async () => {
+    let specificEx: string | undefined = await vscode.window.showQuickPick(
+      Object.keys(examplesJSON)
+    );
+
+    if (specificEx) {
+      specificExGlobal = specificEx;
+      if (specificEx.endsWith(".env") || specificEx.endsWith(".l2")) {
+        vscode.window.activeTextEditor?.edit((builder) => {
+          const doc = vscode.window.activeTextEditor?.document;
+          builder.replace(
+            new vscode.Range(
+              doc!.lineAt(0).range.start,
+              doc!.lineAt(doc!.lineCount - 1).range.end
+            ),
+            examplesJSON[specificExGlobal]
+          );
+        });
+        return;
+      }
 
-  let generateCodeSnippet = new GenerateCodeSnippet();
-  let lama2Examples = vscode.commands.registerCommand(
-    "lama2.Lama2Examples",
-    async () => {
-      let specificEx: string | undefined = await vscode.window.showQuickPick(
-        Object.keys(examplesJSON)
+      let subSpecificEx: string | undefined = await vscode.window.showQuickPick(
+        Object.keys(examplesJSON[specificEx])
       );
 
-      if (specificEx) {
-        specificExGlobal = specificEx;
-        if (specificEx.endsWith(".env") || specificEx.endsWith(".l2")) {
-          vscode.window.activeTextEditor?.edit((builder) => {
-            const doc = vscode.window.activeTextEditor?.document;
-            builder.replace(
-              new vscode.Range(
-                doc!.lineAt(0).range.start,
-                doc!.lineAt(doc!.lineCount - 1).range.end
-              ),
-              examplesJSON[specificExGlobal]
-            );
-          });
-          return;
-        }
-
-        let subSpecificEx: string | undefined =
-          await vscode.window.showQuickPick(
-            Object.keys(examplesJSON[specificEx])
+      if (subSpecificEx) {
+        subSpecificExGlobal = subSpecificEx;
+        vscode.window.activeTextEditor?.edit((builder) => {
+          const doc = vscode.window.activeTextEditor?.document;
+          builder.replace(
+            new vscode.Range(
+              doc!.lineAt(0).range.start,
+              doc!.lineAt(doc!.lineCount - 1).range.end
+            ),
+            examplesJSON[specificExGlobal][subSpecificExGlobal]
           );
-
-        if (subSpecificEx) {
-          subSpecificExGlobal = subSpecificEx;
-          vscode.window.activeTextEditor?.edit((builder) => {
-            const doc = vscode.window.activeTextEditor?.document;
-            builder.replace(
-              new vscode.Range(
-                doc!.lineAt(0).range.start,
-                doc!.lineAt(doc!.lineCount - 1).range.end
-              ),
-              examplesJSON[specificExGlobal][subSpecificExGlobal]
-            );
-          });
-          return;
-        }
+        });
+        return;
       }
     }
-  );
+  });
+}
 
-  // Level1 command pallette
+function isDefault(defaultClient: string, client: string) {
```

Comment:
What is this function about? not able to infer from context.

Reply:
Maybe it should go into a group of functions in a different file. My best guess is it's related to the code generation library selection.

Reply:
Yeah moved everything to new files.